woocommerce/woocommerce
# Fix tests due to PHP 8.1 static variable breaking change

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





PHP 8.1 version introduced a breaking change regarding the use of static variables in inherited methods. The inherited method will now share static variables with the parent method, meaning they now behave the same way as static properties.
This breaking change is making a test fail when running in version 8.1. This PR fixes that failure to be able to do this change: https://github.com/woocommerce/woocommerce/pull/53700

Note: this code is part of the old DI container infrastructure that we stopped using as of WooCommerce 9.5 and will be removed soon (in WC 10.0).

For more context: p7bje6-6EX-p2



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Locally, checkout to `trunk` and change the `phpVersion` to `8.1` in this file: https://github.com/woocommerce/woocommerce/blob/2e078af125129d1ef6fee752afad8419a2b8366d/plugins/woocommerce/.wp-env.json#L3
2. Run `pnpm env:restart`.
3. Run `pnpm --filter="@woocommerce/plugin-woocommerce" test:php:env --filter="ExtendedContainerTest|AbstractInterfaceServiceProviderTest"` and make sure you see the `test_all_classes_implementing_an_interface_are_registered_correctly` test fail.
4. Checkout to this PR branch and run again `pnpm --filter="@woocommerce/plugin-woocommerce" test:php:env --filter="ExtendedContainerTest|AbstractInterfaceServiceProviderTest"`.
5. Make sure all tests are passing.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Fix tests due to PHP 8.1 static variable breaking change.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
